jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/actions/PurgeQueueAction.kt

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.sqs.actions
+
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.project.DumbAware
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import software.amazon.awssdk.services.sqs.SqsClient
+import software.amazon.awssdk.services.sqs.model.PurgeQueueInProgressException
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.core.utils.warn
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.explorer.actions.SingleResourceNodeAction
+import software.aws.toolkits.jetbrains.services.sqs.SqsQueueNode
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
+import software.aws.toolkits.jetbrains.utils.notifyError
+import software.aws.toolkits.jetbrains.utils.notifyInfo
+import software.aws.toolkits.resources.message
+
+class PurgeQueueAction :
+    SingleResourceNodeAction<SqsQueueNode>(message("sqs.purge_queue")),
+    CoroutineScope by ApplicationThreadPoolScope("PurgeQueueAction "),
+    DumbAware {
+    override fun actionPerformed(selected: SqsQueueNode, e: AnActionEvent) {
+        val project = selected.nodeProject
+        val client: SqsClient = project.awsClient()
+        launch {
+            try {
+                client.purgeQueue { it.queueUrl(selected.queueUrl) }
```

Comment:
Should we put the table into busy mode and say purge is in progress? That way they can't alter anything while it is purging

Reply:
Like what happens when you poll while it is purging?

Reply:
The SQS API is weird, it's valid to poll during a purge and you might still get results.
It's also valid to send during a purge, which might or might not actually put it in the queue. 
Delete is also weird in that a delete might not actually delete.

So, I don't know if we actually want to do anything else.

Reply:
welp